nmtwizard/framework.py

```diff
@@ -641,14 +643,17 @@ def serve_wrapper(self, config, model_path, host, port, gpuid=0):
             backend_info_fn=self.backend_info,
             rebatch_request=not self.has_own_request_batching)
 
-    def preprocess(self, config, storage):
+    def preprocess(self, config, storage, output=None):
         logger.info('Starting preprocessing data ')
         start_time = time.time()
 
         local_config = self._finalize_config(config)
-        outputs = self._generate_training_data(local_config)
+        outputs = self._build_data(local_config)
```

Comment:
As discussed, this will activate the merge for the simple preprocess action as well.
Maybe we can pass optional `output` argument and only activate the merge when it's not None ?